File: mobile_flutter/lib/screens/sharing_center_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../api/client.dart';
import '../config/app_icons.dart';
import '../state/language_controller.dart';
import '../state/theme_controller.dart';

const Map<String, (String vi, String en)> _kSourceTypeLabels = {
  'email_summary': ('Email', 'Email'),
  'calendar_event': ('Lịch', 'Calendar'),
  'note': ('Ghi chú', 'Note'),
  'document_reference': ('Tài liệu', 'Document'),
};

/// "Trung tam chia se" (Sharing Center, Phase 4): everything the caller has
/// personally shared into any Business workspace, with a revoke action.
/// Reached from Settings. Not workspace-scoped -- GET /api/user/sharing
/// spans every workspace the caller belongs to. Mirrors the web/RN Sharing
/// Center screens built on the same routes/sharing.py endpoints.
class SharingCenterScreen extends StatefulWidget {
  const SharingCenterScreen({super.key});

  @override
  State<SharingCenterScreen> createState() => _SharingCenterScreenState();
}

class _SharingCenterScreenState extends State<SharingCenterScreen> {
  List<Map<String, dynamic>> artifacts = [];
  bool loading = false;

  @override
  void initState() {
    super.initState();
    _load();
  }

  Future<void> _load() async {
    setState(() => loading = true);
    try {
      final data = await apiGet('/user/sharing');
      if (data is Map && data['success'] == true) {
        artifacts = List<Map<String, dynamic>>.from(
          ((data['artifacts'] as List?) ?? []).map((a) => Map<String, dynamic>.from(a as Map)),
        );
      }
    } catch (_) {}
    if (mounted) setState(() => loading = false);
  }

  Future<void> _revoke(Map<String, dynamic> artifact) async {
    final t = context.read<LanguageController>().t;
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (ctx) => AlertDialog(
        title: Text(t('Thu hồi chia sẻ này?', 'Revoke this share?')),
        content: Text(t(
          'Không gian sẽ không còn thấy nội dung này nữa.',
          'The workspace will no longer see this content.',
        )),
        actions: [
          TextButton(onPressed: () => Navigator.pop(ctx, false), child: Text(t('Hủy', 'Cancel'))),
          TextButton(onPressed: () => Navigator.pop(ctx, true), child: Text(t('Thu hồi', 'Revoke'))),
        ],
      ),
    );
    if (confirmed != true) return;
    try {
      await apiDelete('/workspaces/${artifact['workspace_id']}/shared-artifacts/${artifact['id']}');
      await _load();
    } catch (_) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text(t('Không thu hồi được', 'Could not revoke'))),
        );
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    final colors = context.watch<ThemeController>().colors;
    final t = context.watch<LanguageController>().t;

    return Scaffold(
      backgroundColor: colors.background,
      body: SafeArea(
        child: Column(
          children: [
            Padding(
              padding: const EdgeInsets.fromLTRB(4, 6, 20, 6),
              child: Row(
                children: [
                  IconButton(icon: Icon(AppIcons.emailBack, color: colors.text), onPressed: () => Navigator.pop(context)),
                  Expanded(
                    child: Text(t('Trung tâm chia sẻ', 'Sharing Center'), style: TextStyle(color: colors.text, fontWeight: FontWeight.w700, fontSize: 17)),
                  ),
                ],
              ),
            ),
            Expanded(
              child: RefreshIndicator(
                onRefresh: _load,
                child: ListView(
                  padding: const EdgeInsets.fromLTRB(16, 4, 16, 24),
                  children: [
                    if (loading)
                      const Padding(padding: EdgeInsets.symmetric(vertical: 40), child: Center(child: CircularProgressIndicator()))
                    else if (artifacts.isEmpty)
                      Padding(
                        padding: const EdgeInsets.symmetric(vertical: 40),
                        child: Text(
                          t("Bạn chưa chia sẻ nội dung nào.", "You haven't shared anything yet."),
                          textAlign: TextAlign.center,
                          style: TextStyle(color: colors.textMuted, fontSize: 13),
                        ),
                      )
                    else
                      ...artifacts.map((artifact) {
                        final sourceType = artifact['source_type'] as String? ?? '';
                        final sourceLabel = _kSourceTypeLabels[sourceType];
                        return Container(
                          margin: const EdgeInsets.only(bottom: 8),
                          padding: const EdgeInsets.all(12),
                          decoration: BoxDecoration(
                            color: colors.panelSoft,
                            borderRadius: BorderRadius.circular(12),
                            border: Border.all(color: colors.border),
                          ),
                          child: Row(
                            children: [
                              Expanded(
                                child: Column(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    Text(artifact['title'] as String? ?? '', style: TextStyle(color: colors.text, fontWeight: FontWeight.w600, fontSize: 13)),
                                    const SizedBox(height: 2),
                                    Text(
                                      '${artifact['workspace_name'] ?? ''} · ${sourceLabel != null ? t(sourceLabel.$1, sourceLabel.$2) : sourceType}',
                                      style: TextStyle(color: colors.textMuted, fontSize: 11),
                                    ),
                                  ],
                                ),
                              ),
                              TextButton(
                                onPressed: () => _revoke(artifact),
                                child: Text(t('Thu hồi', 'Revoke'), style: TextStyle(color: colors.danger, fontWeight: FontWeight.w600, fontSize: 12.5)),
                              ),
                            ],
                          ),
                        );
                      }),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: mobile/src/screens/SharingCenterScreen.js
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { ActivityIndicator, Alert, Modal, SafeAreaView, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { radius, useTheme } from '../theme/ThemeContext';
import { useLanguage } from '../i18n/LanguageContext';
import { apiDelete, apiGet } from '../api/client';

const SOURCE_TYPE_LABELS = {
  email_summary: 'Email',
  calendar_event: 'Lịch',
  note: 'Ghi chú',
  document_reference: 'Tài liệu',
};

// "Trung tam chia se" (Sharing Center, Phase 4): everything the caller has
// personally shared into any Business workspace, with a revoke action.
// Reached from Settings. Not workspace-scoped -- GET /api/user/sharing
// spans every workspace the caller belongs to.
export default function SharingCenterScreen({ visible, onClose }) {
  const { colors } = useTheme();
  const { t } = useLanguage();
  const styles = useMemo(() => makeStyles(colors), [colors]);

  const [artifacts, setArtifacts] = useState([]);
  const [loading, setLoading] = useState(false);

  const load = useCallback(async () => {
    setLoading(true);
    try {
      const data = await apiGet('/user/sharing');
      if (data?.success) setArtifacts(data.artifacts || []);
    } catch { /* keep whatever was already shown */ }
    setLoading(false);
  }, []);

  useEffect(() => {
    if (visible) load();
  }, [visible, load]);

  const revoke = (artifact) => {
    Alert.alert(
      t('Thu hồi chia sẻ này?', 'Revoke this share?'),
      t(
        'Không gian sẽ không còn thấy nội dung này nữa.',
        'The workspace will no longer see this content.'
      ),
      [
        { text: t('Hủy', 'Cancel'), style: 'cancel' },
        {
          text: t('Thu hồi', 'Revoke'),
          style: 'destructive',
          onPress: async () => {
            try {
              await apiDelete(`/workspaces/${artifact.workspace_id}/shared-artifacts/${artifact.id}`);
              await load();
            } catch (error) {
              Alert.alert(t('Không thu hồi được', 'Could not revoke'), error.message);
            }
          },
        },
      ]
    );
  };

  return (
    <Modal visible={visible} animationType="slide" onRequestClose={onClose}>
      <SafeAreaView style={styles.root}>
        <View style={styles.header}>
          <TouchableOpacity onPress={onClose} hitSlop={{ top: 8, bottom: 8, left: 8, right: 8 }}>
            <Ionicons name="arrow-back" size={22} color={colors.text} />
          </TouchableOpacity>
          <Text style={styles.headerTitle}>{t('Trung tâm chia sẻ', 'Sharing Center')}</Text>
        </View>

        <ScrollView contentContainerStyle={styles.body}>
          {loading ? (
            <ActivityIndicator style={{ marginVertical: 40 }} color={colors.primary} />
          ) : artifacts.length ? (
            artifacts.map((artifact) => (
              <View key={artifact.id} style={styles.row}>
                <View style={{ flex: 1 }}>
                  <Text style={styles.title}>{artifact.title}</Text>
                  <Text style={styles.meta}>
                    {artifact.workspace_name} · {SOURCE_TYPE_LABELS[artifact.source_type] || artifact.source_type}
                  </Text>
                </View>
                <TouchableOpacity onPress={() => revoke(artifact)}>
                  <Text style={styles.revokeText}>{t('Thu hồi', 'Revoke')}</Text>
                </TouchableOpacity>
              </View>
            ))
          ) : (
            <Text style={styles.emptyText}>{t('Bạn chưa chia sẻ nội dung nào.', "You haven't shared anything yet.")}</Text>
          )}
        </ScrollView>
      </SafeAreaView>
    </Modal>
  );
}

function makeStyles(colors) {
  return StyleSheet.create({
    root: { flex: 1, backgroundColor: colors.background },
    header: { flexDirection: 'row', alignItems: 'center', gap: 14, paddingHorizontal: 16, paddingVertical: 12 },
    headerTitle: { color: colors.text, fontFamily: 'Poppins_700Bold', fontSize: 17 },
    body: { paddingHorizontal: 16, paddingBottom: 32, gap: 10 },
    row: {
      flexDirection: 'row',
      alignItems: 'center',
      padding: 12,
      borderRadius: radius.control,
      backgroundColor: colors.panelSoft,
      borderWidth: 1,
      borderColor: colors.border,
    },
    title: { color: colors.text, fontFamily: 'Poppins_600SemiBold', fontSize: 13 },
    meta: { color: colors.textMuted, fontFamily: 'Poppins_400Regular', fontSize: 11.5, marginTop: 2 },
    revokeText: { color: colors.danger, fontFamily: 'Poppins_600SemiBold', fontSize: 12.5 },
    emptyText: { color: colors.textMuted, fontFamily: 'Poppins_400Regular', fontSize: 13, textAlign: 'center', marginTop: 40 },
  });
}
